File: core/src/main/java/de/qabel/core/repository/sqlite/migration/MigrationFailedException.java
package de.qabel.core.repository.sqlite.migration;

import de.qabel.core.repository.sqlite.MigrationException;

public class MigrationFailedException extends MigrationException {
    private AbstractMigration failedMigration;

    public MigrationFailedException(AbstractMigration failedMigration, String message) {
        super(message);
        this.failedMigration = failedMigration;
    }

    public MigrationFailedException(AbstractMigration failedMigration, String message, Throwable cause) {
        super(message, cause);
        this.failedMigration = failedMigration;
    }

    public AbstractMigration getFailedMigration() {
        return failedMigration;
    }
}

File: core/src/main/java/de/qabel/core/repository/sqlite/migration/MigrationFailedException.kt
package de.qabel.core.repository.sqlite.migration

import de.qabel.core.repository.sqlite.MigrationException

class MigrationFailedException : MigrationException {
    var failedMigration: AbstractMigration? = null
        private set

    constructor(failedMigration: AbstractMigration, message: String) : super(message) {
        this.failedMigration = failedMigration
    }

    constructor(failedMigration: AbstractMigration, message: String, cause: Throwable) : super(message, cause) {
        this.failedMigration = failedMigration
    }
}
